source/jmap/types.d

```diff
@@ -119,66 +112,46 @@ struct Session {
         import std.algorithm : canFind;
 
         if (activeAccountId_.length == 0) {
-            enforce(accounts.keys.length == 1,
-                    format!"multiple accounts - [%s] - and you must call setActiveAccount to pick one"
-                    (accounts.keys.join(",")));
-            this.activeAccountId_ = accounts.keys.front;
+            if (accounts.keys.length != 1)
+                throw new MirException("multiple accounts - ", accounts.keys, " - and you must call setActiveAccount to pick one");
+            this.activeAccountId_ = accounts.keys[0];
         }
-
-        enforce(accounts.keys.canFind(activeAccountId_,
-                format!"active account ID is set to %s but it is not found amongst account IDs: [%s]"
-                (activeAccountId_, accounts.keys.join(","))));
+        else if (!accounts.keys.canFind(activeAccountId_))
+            throw new MirException("active account ID is set to ", activeAccountId_, " but it is not found amongst account IDs: ", activeAccountId_);
         return activeAccountId_;
     }
 
-    string[] listCapabilities() {
+    const(string)[] listCapabilities() const {
         return capabilities.keys;
     }
 
-    string[] listAccounts() {
-        import std.algorithm : map;
-        import std.array : array;
-        return accounts.keys.map!(key => accounts[key].name).array;
+    string[] listAccounts() const {
+        return accounts.values.member!"name".as!string.array;
     }
 
     Account getActiveAccountInfo() {
-        import std.exception : enforce;
-        auto p = activeAccountId() in accounts;
-        enforce(p !is null, "no currently active account");
-        return *p;
+        return *enforce!"no currently active account"(activeAccountId() in accounts);
     }
 
     @SILdoc("set active account - name is the account name, not the id")
     Session setActiveAccount(string name) {
-        import std.format : format;
-        import std.exception : enforce;
-        import std.format : format;
 
-        foreach (kv; accounts.byKeyValue) {
-            if (kv.value.name == name) {
-                this.activeAccountId_ = kv.key;
+        foreach (i, ref value; accounts.values) {
+            if (value.name == name) {
+                this.activeAccountId_ = accounts.keys[i];
                 return this;
             }
         }
-        throw new Exception(format!"account %s not found"(name));
+        throw new MirException("account ", name, " not found");
     }
 
-
-    void finalizeDeserialization(Asdf data) {
-        import asdf : deserialize, Asdf;
-
-        foreach (el; data["capabilities"].byKeyValue)
-            capabilities[el.key] = el.value.get!Asdf(Asdf.init);
-        this.coreCapabilities = deserialize!SessionCoreCapabilities(capabilities["urn:ietf:params:jmap:core"]);
+    void serdeFinalize() {
+        this.coreCapabilities = capabilities["urn:ietf:params:jmap:core"].serde!SessionCoreCapabilities;
     }
-
+version(SIL):
```

Comment:
why is post suddenly `version(SIL)`?

Also, there are a few remaining version conditionals that are redundant if this one stays.

Reply:
The whole JMAP is `version(SIL)` in master. So I am removing it part by part. There is no reason to make additional version cleanup because they are finally removed completely in the next MRs.